tools/src/main/scala/scala/scalanative/codegen/LLVMDebugInformation.scala

```diff
@@ -0,0 +1,375 @@
+package scala.scalanative.codegen
+package llvm
+
+import DebugInformationSection._
+import LLVMDebugInformation._
+import Val.Str
+import Val.Num
+import scala.scalanative.nir.{Position, Global}
+import scala.reflect.ClassTag
+
+private[llvm] trait GenIdx {
+  def id(tok: DebugInformationSection.Token): Int
+  def gen: DebugInformationSection.Token
+}
+private[llvm] object GenIdx {
+  def create = new GenIdx {
+    private val mp = collection.mutable.Map.empty[Token, Int]
+    override def id(tok: Token): Int = mp(tok)
+    override def gen: Token = {
+      val newVal = new Token
+      val newIdx = mp.size
+
+      mp.synchronized {
```

Comment:
Each GenIdx instance is accessible by a single thread (AbstractCodeGen.Impl instance) so synchronization should not be needed